packages/vc-verification/src/models/cache-interfaces.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  IRoleDefinitionV2,
+  RoleCredentialSubject,
+} from '@energyweb/credential-governance';
+import { VerifiableCredential } from '@ew-did-registry/credentials-interface';
+import { RoleEIP191JWT } from './credentials-types';
+import { IDIDDocument } from '@ew-did-registry/did-resolver-interface';
+
+/**
+ * An Interface to cache DID Document from Blockchain and SSI-Hub
+ */
+export interface IDIDDocumentCache {
+  /**
+   * Stores DID Document
+   * @param did user DID
+   * @param data DIDDocument of the user (DID)
+   */
+  setDIDDocument(did: string, data: IDIDDocument): void;
+
+  /**
+   * Returns cached DID Document (services)
+   * @param did user DID
+   */
+  getDIDDocument(did: string): IDIDDocument | undefined;
+}
+
+/**
+ * An Interface to cache role credential from Blockchain and SSI-Hub
+ */
+export interface IRoleCredentialCache {
+  /**
+   * Stores role credential
+   * @param did user DID
+   * @param role namespace
+   * @param data Credential of the user for the namespace
+   */
+  setRoleCredential(
```

Comment:
I would also chose more descriptive names, so that when you see method you understand on which object iit is called
```suggestion
  cacheRoleCredential(
```

Reply:
IMO, think `set` is actually clearer than `cache`, but I would leave it up to you @nichonien